lib/after_commit_everywhere.rb

```diff
@@ -91,7 +92,12 @@ def register_callback(connection:, name:, no_tx_action:, callback:)
 
     def in_transaction?(connection)
       # service transactions (tests and database_cleaner) are not joinable
-      connection.transaction_open? && connection.current_transaction.joinable?
+      is_current_transaction_joinable = if Gem.loaded_specs["activerecord"].version < Gem::Version.create('4.2')
```

Comment:
Is there any reason you're not using `ActiveRecord::VERSION` (e.g. `Gem::Version.new(ActiveRecord::VERSION::STRING`)?